File: Hash Map/LET_no1624_Largest Substring Between Two Equal Characters.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        maxLen, rec = -1, {}
        for i, ch in enumerate(s):
            maxLen = max(maxLen, i - rec.setdefault(ch, i) - 1)
        return maxLen
        
        # res, rec = -1, {}
        # for i in range(len(s)):
        #     if s[i] not in rec: rec[s[i]] = i
        #     else: res = max(res, i - rec[s[i]] - 1)
        # return res

File: Data Structure/Hash/Easy/LET_no1624_Largest Substring Between Two Equal Characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        int res = -1;
        unordered_map<char, vector<int>> memo;
        for (int i = 0; i < s.size(); i++) {
            if (memo.count(s[i]))
                res = max(res, i - memo[s[i]].back() - 1);
            else
                memo[s[i]].push_back(i);
        }

        return res;
    }
};